karma.conf.js

```diff
@@ -29,23 +29,31 @@ module.exports = function (config) {
         colors: true,
         logLevel: config.LOG_INFO,
         autoWatch: false,
+        exclude: ['web3'],
         browsers: [
             'ChromeHeadless',
             'FirefoxHeadless'
         ],
+        browserify: {
+            configure: function(bundle) {
+                bundle.on('prebundle', function() {
+                    bundle.external('../../packages/web3');
+                });
+            }
+        },
```

Comment:
There is a [parcel plugin][1] for Karma, if that less likely than browserify to mask anything. 

What does the "prebundle" step do? In the existing headless browser tests, Web3 gets directly required from the min in the test helpers to verify that nothing was corrupted / broken there. Does this change that?  

[1]: https://github.com/valotas/karma-parcel

Reply:
> What does the "prebundle" step do? In the existing headless browser tests, Web3 gets directly required from the min in the test helpers to verify that nothing was corrupted / broken there. Does this change that?

It was bundling the require statement (``require('../../packages/web3');``) also when not used that’s why I've added the ``external`` method call for the bundling process. It is added only to be sure that it will never use the non-minified bundle for e2e tests.

Reply:
I've tested this by commenting out the karma configs `exclude`, `prebundle` *and* the non-window conditional `require` in the helpers (so browserify ignores it) and it's clear `window.Web3` is used / tests pass.  

But I'm a little confused because the last time I checked this I think removing the babel/polyfill alone allowed everything to work without extra configuration. Now without these extra things I get an error complaining about not being able to find `is-buffer`. 

In the end it doesn't seem relevant though.

Resolving...